cdap-docs/developers-manual/source/advanced/etl.rst

```diff
@@ -0,0 +1,85 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _advanced-custom-etl:
+
+===============================================
+Creating Custom ETL Applications and Components
```

Comment:
I see we have "Creating Application Templates" also below. So we should mention that this page, documents both. Or we can split it into two pages.